/*
 * Protocol Buffer definition for ProductService gRPC API
 * (Copy from ProductService for client code generation)
 */

syntax = "proto3";

package product;

service ProductService {
    rpc GetProductInfo (ProductInfoRequest) returns (ProductInfoResponse);
    rpc CheckAvailability (AvailabilityRequest) returns (AvailabilityResponse);
}

message ProductInfoRequest {
    int32 product_id = 1;
    string requesting_service = 2;
}

message ProductInfoResponse {
    bool exists = 1;
    ProductInfo product_info = 2;
    string error_message = 3;
}

message ProductInfo {
    int32 id = 1;
    string name = 2;
    string description = 3;
    string price = 4;
    int32 inventory_count = 5;
    bool is_available = 6;
}

message AvailabilityRequest {
    int32 product_id = 1;
    int32 quantity = 2;
    string requesting_service = 3;
}

message AvailabilityResponse {
    bool available = 1;
    int32 available_quantity = 2;
    string error_message = 3;
}
